memcache.go

```diff
@@ -83,34 +85,23 @@ func (c *Client) GetString(k string) (string, bool) {
 func (i *Item) String() (string, error) {
 	switch i.Flags {
 	case FLAG_PICKLE:
-		// Note: unicode objects get pickled, but parsing them is straight forward
-		//
-		// parsing is based on the following and only checks for protocol version 2
-		// - http://spootnik.org/entries/2014/04/05_diving-into-the-python-pickle-format.html
-		// - http://www.hydrogen18.com/blog/reading-pickled-data-in-go.html
-		// - https://github.com/hydrogen18/stalecucumber
-		//
-		// record format: (10 bytes in addition to the string)
-		// 2 pickle pre-amble - 0x80, 0x2 (pickle flag and version)
-		// 1 byte unicode opcode - 0x58
-		// 4 byte size - little endian
-		// ...
-		// 2 byte BINPUT 1 - 0x71, 0x1
-		// 1 byte stop opcode  - 0x2e
-		unicodePreamble := []byte{0x80, 0x2, 0x58}
-		if bytes.HasPrefix(i.Value, unicodePreamble) && len(i.Value) >= 10 {
-			size := binary.LittleEndian.Uint32(i.Value[3:])
-			if size+10 == uint32(len(i.Value)) {
-				return string(i.Value[7 : 7+size]), nil
-			}
+		s, err := unpickle(string(i.Value))
+		if err == nil {
+			return s.(string), nil
 		}
 	case FLAG_NONE:
+		if bytes.HasPrefix(i.Value, []byte{0x80, 0x2}) {
```

Comment:
I think it would be good to have a `TestItem_String` and `TestItem_Int64` that captures example values `(flags, value)`  to document the types of values we expect to handle.